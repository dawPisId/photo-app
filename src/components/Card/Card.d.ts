interface InputProps {
    name: string,
    url: string,
    action: () => void
  }
  