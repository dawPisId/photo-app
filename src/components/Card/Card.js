import { Label, Layout } from "./Card.styled";

import React from "react";

const Card = ({ name, url, action }) => (
  <Layout onClick={action}>
    <img src={url} alt={name}></img>
    <Label>{name}</Label>
  </Layout>
);

export default Card;
