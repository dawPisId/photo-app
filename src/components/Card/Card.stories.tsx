import { ComponentMeta, ComponentStory } from "@storybook/react";

import Card from "./Card";
import Grid from "../Grid/Grid";
import React from "react";

export default {
  title: "Photo App/Card",
  component: Card,
} as ComponentMeta<typeof Card>;

const Template: ComponentStory<typeof Card> = (args) => (
  <Grid>
    <Card {...args} />
  </Grid>
);

export const Main = Template.bind({});
Main.args = {
  name: "Card",
  url: "https://via.placeholder.com/600/92c952",
  action: () => null,
};
