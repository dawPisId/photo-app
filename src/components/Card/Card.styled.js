import styled from "styled-components";

export const Layout = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: white;
  cursor: pointer;
  border-radius: 5px;
  cursor: pointer;
  box-shadow: 0px 5px 5px grey;
  height: 20rem;
  overflow: hidden;
  img {
    border-top: 5px;
    min-width: 100%;
    height: 80%;
    display: block;
  }
`;

export const Label = styled.span`
  height: 10%;
  margin: 1rem 1rem 0.5rem 1rem;
  overflow: scroll;
  -ms-overflow-style: none;
  font-size: 1rem;
  font-family: "Roboto", sans-serif;
  &::-webkit-scrollbar {
    width: 0 !important;
  }
`;
