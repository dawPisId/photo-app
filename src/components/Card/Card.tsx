import { Label, Layout } from "./Card.styled";

import React from "react";

interface InputProps {
  name: string,
  url: string,
  action: () => void
}

const Card = ({ name, url, action }:InputProps) => (
  <Layout onClick={action}>
    <img src={url} alt={name}></img>
    <Label>{name}</Label>
  </Layout>
);

export default Card;
