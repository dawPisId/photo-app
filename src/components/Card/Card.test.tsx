import Card from "./Card";
import React from "react";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Card
        name="one"
        url="https://via.placeholder.com/600/92c952"
        action={() => null}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
