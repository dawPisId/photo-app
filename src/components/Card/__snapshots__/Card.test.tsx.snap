// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdnxRM kJPHt"
  onClick={[Function]}
>
  <img
    alt="one"
    src="https://via.placeholder.com/600/92c952"
  />
  <span
    className="sc-gtsrHT dbOeOW"
  >
    one
  </span>
</div>
`;
